### DISCUSSION TITLE
Custom fonts

### DISCUSSION THREAD

**Discussion by dadiggi** [gickowtf/pixoo-homeassistant#51] [Ideas]
Hi!
I've noticed that fonts are currently embedded within the code (specifically in _font.py). This setup allows for some degree of customization, such as adding special characters or letters not present in the English language to existing fonts. However, these customizations are overwritten with each update, posing a challenge for maintainability and user-specific adaptations.
To improve this, would it be possible to implement a logic where additional fonts are dynamically loaded from a user-defined directory or file?
This approach would not only preserve custom font adjustments across updates but also enhance the flexibility of font management. The fonts could retain the same format as the current embedded ones for compatibility.

**Comment by SB2DD** [comment]
I'm not against this, but if you want, in the meantime, you're free to open a PR with your non-english characters!

**Reply by SB2DD** [reply]
Whilst we're at it, maybe we could also do something similar for custom pages? Like the PV or loading bar one?

**Reply by gickowtf** [reply]
it's definitely real work to design characters. i deliberately left out german characters like äüö and also limited special characters to the minimum. how exactly do you mean with the PV page that only has numbers?

**Reply by dadiggi** [reply]
Yes, unfortunately I know that designing fonts is a tedious job, but just to avoid any misunderstanding, I didn't mean that you create them, but rather, if it's not too demanding for development, to allow users to create and use their own set of characters if they wish.

@Mrredstone5230 Thanks for the PR suggestion. However, I believe it's preferable to maintain "standard" fonts within the basic set of characters to ensure consistency. For the moment, it's still easy enough to adjust the code for the purpose of adding custom characters. 😉

**Reply by SB2DD** [reply]
> how exactly do you mean with the PV page that only has numbers?

I was thinking that users might be able to create their own pages in python (therefore being able to define their own page_type) And if this font idea actually gets made, maybe we could allow this custom page feature too.
